docker/Dockerfile

```diff
@@ -1,44 +1,36 @@
-FROM python:3.7.2-alpine3.8
-
-ENV PYTHONUNBUFFERED=1
-ENV PYTHONDONTWRITEBYTECODE=1
+FROM python:3.7.3-slim as base
 ENV PATH="/root/.poetry/bin:$PATH"
 ENV FLASK_APP=search
-
 WORKDIR /srv/app
-
-RUN rm -rf /var/cache/apk/* && \
-    rm -rf /tmp/* && \
-    apk update && \
-    apk add --no-cache \
-    bash \
-    grep \
-    libxslt-dev
-
-RUN apk add --no-cache --virtual .build-dependencies \
-    curl \
-    && \
-    curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python && \
-    poetry config settings.virtualenvs.create false && \
-    apk del .build-dependencies
-
 COPY ./pyproject.toml ./poetry.lock ./
-
-ARG env
-ENV FLASK_ENV=${env}
-
-ARG dev
-ARG simulate_removal
-RUN apk add --no-cache --virtual .build-dependencies \
-    gcc \
-    git \
-    linux-headers \
-    musl-dev \
-    && \
-    poetry install ${dev} && \
-    apk del ${simulate_removal} .build-dependencies
-
+RUN set -ex \
+    && apt-get update -yq \
+    && apt-get install -yq --no-install-recommends curl build-essential\
+    && curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python \
+    && poetry config settings.virtualenvs.create false \
+    && poetry install --no-dev \
```

Comment:
Is it possible to separate the installation of `poetry` from `poetry install`?

The order I usually see is:
1. install package manager
2. `COPY` lock files and similar
3. run package manager `install` command
4. `COPY` everything else

This way `1` almost never needs to be executed again, if the Docker layer cache is there.

Reply:
I did this earlier and you asked: `why not a single RUN command with the one before?`

Reply:
i'll update again